lib/experimental/new_embed.dart

```diff
@@ -178,6 +178,10 @@ class NewEmbed {
     });
 
     executionSvc.testResults.listen((result) {
+      if (result.messages.isEmpty &&
```

Comment:
There's another check a few lines down that involves messages being empty, so it might be cleaner to combine them. Something like:

```dart
if (result.message.isEmtpy) {
  result.messages.add(result.success ? 'Test passed!' : 'Test failed.');
}
```

Then you could drop line 186 entirely.